src/libraries/System.Net.Primitives/tests/UnitTests/CookieContainerTest.cs

```diff
@@ -762,5 +763,66 @@ private void VerifyGetCookies(CookieContainer cc1, Uri uri, Cookie[] expected)
                 Assert.Equal(c1.Value, c2.Value);
             }
         }
+
+        [Theory]
+        [InlineData("http://localhost:5000", "localhost", null)]
+        [InlineData("http://localhost:5000", "localhost", 0)]
+        [InlineData("http://localhost:5000", "localhost", 1)]
+        [InlineData("http://localhost:5000", "localhost", 100)]
+        [InlineData("http://localhost:5000", ".localhost", null)]
+        [InlineData("http://localhost:5000", ".localhost", 0)]
+        [InlineData("http://localhost:5000", ".localhost", 1)]
+        [InlineData("http://localhost:5000", ".localhost", 100)]
+        [InlineData("http://example.com", "example.com", null)]
+        [InlineData("http://example.com", "example.com", 0)]
+        [InlineData("http://example.com", "example.com", 1)]
+        [InlineData("http://example.com", "example.com", 100)]
+        [InlineData("http://example.com", ".example.com", null)]
+        [InlineData("http://example.com", ".example.com", 0)]
+        [InlineData("http://example.com", ".example.com", 1)]
+        [InlineData("http://example.com", ".example.com", 100)]
+        [InlineData("http://test.example.com", "test.example.com", null)]
+        [InlineData("http://test.example.com", "test.example.com", 0)]
+        [InlineData("http://test.example.com", "test.example.com", 1)]
+        [InlineData("http://test.example.com", "test.example.com", 100)]
+        [InlineData("http://test.example.com", ".test.example.com", null)]
+        [InlineData("http://test.example.com", ".test.example.com", 0)]
+        [InlineData("http://test.example.com", ".test.example.com", 1)]
+        [InlineData("http://test.example.com", ".test.example.com", 100)]
+        public void SetCookies_DomainCheckIgnoresVersion(string url, string domain, int? version)
+        {
+            string cookie = $"my_cookie=my_value; Domain={domain}" + (version != null ? $"; Version = {version}" : "");
+            var uri = new Uri(url);
+            CookieContainer container = new CookieContainer();
+            container.SetCookies(uri, cookie);
+            Assert.Equal(1, container.GetCookies(uri).Count);
+            Assert.Equal(domain, container.GetCookies(uri)[0].Domain);
+        }
+
+        [Theory]
+        [InlineData("http://test.example.com", ".example.com", true)]
```

Comment:
Same as above.

Reply:
I ended up making two tests from one to get rid of bool parameter. I think resulted InlineData is now not big enough to switch to MemberData